esmvaltool/utils/cmorizers/obs/cmorize_obs_HadCRUT3.ncl

```diff
@@ -0,0 +1,137 @@
+; #############################################################################
+; ESMValTool CMORizer for HadCRUT3 data
+; #############################################################################
+;
+; Tier
+;    Tier 2: other freely-available dataset.
+;
+; Source
+;    http://www.metoffice.gov.uk/hadobs/hadcrut3/data/download.html
+;
+; Last access
+;    20190221
+;
+; Download and processing instructions
+;    Download the HadCRUT3v.nc file (variance adjusted dataset).
+;
+; Caveats
+;    The HadCRUT3v variance-adjusted dataset for tas is actually the anomaly
+;    with respect to the period 1958-2001.
+;
+; Modification history
+;    20190221-A_righ_ma: adapted to v2 and renamed to HadCRUT3.
+;    20150330-A_righ_ma: updated paths and global attributes.
+;    20140311-A_senf_da: written.
+;
+; #############################################################################
+loadscript(getenv("esmvaltool_root") + "/utils/cmorizers/obs/interface.ncl")
+
+begin
+
+  ; Script name (for logger)
+  DIAG_SCRIPT = "cmorize_obs_HadCRUT3.ncl"
+
+  ; Source name
+  OBSNAME = "HadCRUT3"
+
+  ; Tier
+  TIER = 2
+
+  ; Period
+  YEAR1 = 1850
+  YEAR2 = 2013
+
+  ; Selected variable (standard name)
+  VAR = "tas"
```

Comment:
I thought HadCRUT contains temperature anomalies. If that would be the case, calling the output variables tas is missleading. An option might be to add the HadCRUT climatology in order to convert anomalies back to absolute values. Another option could be to use another variable name, e.g. tasa.

Reply:
The recipe/diagnostic would need to be changed accordingly. This is used in `recipe_cox18nature.yml` at the moment. @schlunma is the renaming to `tasa` fine with you?

Reply:
Yeah, sure. I will adapt the script accordingly, should not be a big issue.

Reply:
Now HadCRUT3 contains `tasa` (anomaly) instead of `tas`.
I did the same also for HadCRUT4, which now delivers both `tas` (adding the climatology as before) and `tasa`. 

@schlunma please adapt `recipe_cox18nature.yml` and let me know if it works (you will get the path to the data via e-mail).

Reply:
I cannot load `tasa` data:
```
esmvaltool._recipe_checks.RecipeError: Unable to load CMOR table 'OBS' for variable 'tasa' with mip 'Amon'
```
Shouldn't #755 solve this?

Reply:
Indeed it should, at least it worked in all other recipes containing custom variables.
@jvegasbsc are we missing something?

Reply:
Wait, this cannot work unless you add the derivation for `tasa`, since the models do not include this variable. If you calculate the anomaly in the diagnostic, you need to move this calculation to the `derive` module.

Reply:
At the moment, I request `tasa` only for  the `OBS` data, not for the CMIP5 models, so it should work. Deriving it from `tas` will be the next step.

Reply:
@jvegasbsc we need your help!

Reply:
seems a bit dodgy to me since the `RecipeError` object has no member `Unable to load CMOR table...` to raise, you sure there is no primary error that triggers this?

Reply:
also your CMOR_tasa.dat file has `SOURCE: CMIP5 (adapted from tas)`, may be a dumbarse but shouldn't it be just `SOURCE: CMIP5`?

Reply:
here's your problem: in `_recipe.py` l.119:
```
    if derive and table_entry is None:
        custom_table = CMOR_TABLES['custom']
        table_entry = custom_table.get_variable(mip, short_name)
```
if you are not deriving it, it will not read the custom table entry; take out `derive and` and it'll go through

Reply:
I guess this still works only if `derive: true` in the recipe. 
I also thought that #755 had solved this.
@axel-lauer @jvegasbsc can you comment?

Reply:
I also thought this has been solved already (not sure any more in which PR). In my opinion, it is essential that custom variables can be "non-derived" such as, for instance, prStderr. This is all a bit strange since I could read the custom variable prStderr with the version2_development branch (about 1 day old) like this:

      prStderr:
        reference_dataset: TRMM-L3
        mip: Amon
        field: T2Ms
        frequency: mon
        start_year: 1998
        end_year: 2013
        additional_datasets:
          - {dataset: TRMM-L3, project: obs4mips, level: v7, version: 7A, tier: 1}

But maybe I am missing something here...

Reply:
I didn't test it with the `frequency` key though, I also thought this was no longer required.
Let me try...

Reply:
I've tried with `obs550aerStderr`, it does not work:
```
esmvaltool._recipe_checks.RecipeError: Unable to load CMOR table 'OBS' for variable 'od550aerStderr' with mip 'aero'
```
Note that I only have 1 dataset in my test recipe and it is `OBS`.

Reply:
Removing `derive` here

https://github.com/ESMValGroup/ESMValTool/blob/25972c1dbbf3cd3b87b11b412b1dc89a1e281148/esmvaltool/_recipe.py#L119

removes the issue, but we should definitely test if this has unwanted side effects.

However, the file cannot be read, it fails with

```
2019-02-28 17:09:06,905 UTC [39354] ERROR   Failed to run fix_metadata([<iris 'Cube' of height / (m) (scalar cube)>, <iris 'Cube' of Near-Surface Air Temperature Anomaly / (K) (time: 2028; latitude: 36; longitude: 72)>], {'project': 'OBS', 'dataset': 'HadCRUT4', 'short_name': 'tasa', 'cmor_table': 'OBS', 'mip': 'Amon', 'frequency': 'mon'})
ValueError: Cubes were not reduced to one afterfixing: 0: height / (m)                        (scalar cube)
1: Near-Surface Air Temperature Anomaly / (K) (time: 2028; latitude: 36; longitude: 72)
```

Reply:
it should be fine __if__ __all__ derived vars are custom ie they dont have a standard cmor table; support for scalar cubes may be in iris 2.2.0 but we may not have caught up with it - @jvegasbsc ?

Reply:
>  but we should definitely test if this has unwanted side effects

For @axel-lauer this works without any change to the code, I'm trying to find out why. 
We should be careful with changing the code here.

@schlunma if I understand correctly, what you need in `recipe_cox18nature.yml` is to read `tas` from the models and `tasa` from the observations. This case already exists, @axel-lauer is using it to read the standard errors from the obs only, so it has to work for this case as well.

Reply:
Also note that you cannot write a derivation routine for `tasa`, since the anomaly calculation depends on the reference period, which is different from case to case. So you would need to keep this calculation in the diagnostic script.

Reply:
We found a solution, see #899. 
@schlunma can you have a look if it helps?

Reply:
Testing ASAP

Reply:
Yes, it works! :+1:  The issue mentioned above remains, I guess we have to remove the `height` variable in the `HadCRUT4` file.

Reply:
Done! Please try again.

Reply:
Perfect, it works now, thanks! :tada:

Reply:
Great! Then please go ahead and commit the required changes to this recipe. :beer: